### ISSUE TITLE
Confused code in dual-steam logits

### ISSUE THREAD

**Issue by HelloWorldU** [MoonshotAI/Kimi-Audio#142]
l'm currently working on  "Add Kimi-Audio integration for vLLM"(see https://github.com/vllm-project/vllm/pull/21849), while the inferece code is strange that l'm not sure which logits they are handling.
consider no return_dict situation, in  modeling_kimia.py L855-922:
```
        audio_logits = self.lm_head(hidden_states)
        text_logits = self.mimo_output(mimo_hidden_states)

        if not return_dict:
            output = (text_logits, audio_logits) + outputs[2:]
            return output
```

Meanwhile, in external class KimiAudio(kimia.py), after forward():
```
        for i in tqdm.tqdm(
            range(max_new_tokens), desc="Generating tokens", disable=False
        ):
            audio_logits, text_logits, past_key_values = self.alm.forward(
                input_ids=decoder_input_audio_ids,
                text_input_ids=decoder_input_text_ids,
                whisper_input_feature=decoder_input_whisper_feature,
                is_continuous_mask=decoder_is_continuous_mask,
                position_ids=decoder_position_ids,
                past_key_values=past_key_values,
                return_dict=False,
            )
```
The variable naming and ordering inconsistency is causing confusion，also to logits_head

**Comment by HelloWorldU** [comment]
@AlanSwift Could you take a look about this?

**Comment by jpWang** [comment]
@HelloWorldU Hi, I have also trained Kimi-audio recently, and I found that `lm_head` actually generates `text logits`, while `mimo_output` generates `audio logits`. In `MoonshotKimiaModel`, you can find that it uses extra mimo layers to generate `mimo_hidden_states`:
```
# extra 1B audio transformers
self.mimo_layers = nn.ModuleList(
    [MoonshotDecoderLayer(config) for _ in range(config.kimia_mimo_layers)]
)
```
